### PR TITLE
Migrate BottomSheetState to Kotlin

### PR DESCRIPTION
**Pull request by shobhitagarwal1612**
Towards #532 

@scolsen @JSunde  PTAL?

### CODE REVIEW

File: ground/src/main/java/com/google/android/ground/ui/home/HomeScreenFragment.java [hunk 1]
```diff
@@ -346,25 +346,25 @@ public boolean onOptionsItemSelected(@NonNull MenuItem item) {
...
                     R.string.loi_delete_confirmation_dialog_title,
-                    locationOfInterestHelper.getLabel(locationOfInterestToDelete)))
+                    locationOfInterestHelper.getLabel(loi)))
             .setMessage(R.string.loi_delete_confirmation_dialog_message)
             .setPositiveButton(
                 R.string.delete_button_label,
                 (dialog, id) -> {
                   hideBottomSheet();
-                  viewModel.deleteLocationOfInterest(locationOfInterestToDelete.get());
+                  viewModel.deleteLocationOfInterest(loi.get());
```
Comment (gino-m) [inline ground/src/main/java/com/google/android/ground/ui/home/HomeScreenFragment.java:367]: Why is the LOI being deleted? What does that actually do?

File: ground/src/main/java/com/google/android/ground/ui/home/mapcontainer/MapContainerFragment.java [hunk 2]
```diff
@@ -219,21 +219,19 @@ private void moveToNewPosition(Point point) {
   }
 
   private void onBottomSheetStateChange(BottomSheetState state, MapFragment map) {
-    mapContainerViewModel.setSelectedLocationOfInterest(state.getLocationOfInterest());
+    Optional<LocationOfInterest> loi = Optional.ofNullable(state.getLocationOfInterest());
```
Comment (gino-m) [inline ground/src/main/java/com/google/android/ground/ui/home/mapcontainer/MapContainerFragment.java:222]: Not relevant to this PR, but just a reminder: LOIs will soon not be able to be removed/modified from the Android client. There will be a separate flow for editing/deleting points captured as part of the data collection flow.

**Review by gino-m — COMMENTED** [review]
Were you able to test this interactively?

**Review by JSunde — APPROVED** [review]

File: ground/src/main/java/com/google/android/ground/ui/home/mapcontainer/MapContainerFragment.java (hunk 2 above)
Comment (shobhitagarwal1612) [reply ground/src/main/java/com/google/android/ground/ui/home/mapcontainer/MapContainerFragment.java:222]: I'll file a separate bug to clean it up. Thanks for pointing it out.

File: ground/src/main/java/com/google/android/ground/ui/home/HomeScreenFragment.java (hunk 1 above)
Comment (shobhitagarwal1612) [reply ground/src/main/java/com/google/android/ground/ui/home/HomeScreenFragment.java:367]: Earlier, the app provided the capability to remove the feature if you're the owner of the project or if the feature was added by you. This should be removed along with the ui components.

**Comment by shobhitagarwal1612** [comment]
> Were you able to test this interactively?

No, I did not test the app as the renaming is still in progress. Just the unit tests and code health checks.

**Pull request merged**